src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -409,24 +427,51 @@ private static void ThrowFieldIsNoIndexedAndStored(IndexField field)
         throw new InvalidOperationException($"A field `{field.Name}` that is neither indexed nor stored is useless because it cannot be searched or retrieved.");
     }
 
-    private void AssertOrAdjustIndexingOptionForComplexObject(IndexField field)
+    private void AssertIndexingBehaviorForComplexObjectInStaticIndex(IndexField field)
     {
+        Debug.Assert(_index.Type.IsStatic(), $"It is is supposed to be called for static indexes only while we got {_index.Type} index");
+
         if (IgnoreComplexObjectsDuringIndex)
             return;
-        
-        Debug.Assert(field.Indexing != FieldIndexing.No, "field.Indexing != FieldIndexing.No");
 
-        if (_index.GetIndexDefinition().Fields.TryGetValue(field.Name, out var fieldFromDefinition) &&
-            fieldFromDefinition.Indexing != null && 
-            fieldFromDefinition.Indexing != FieldIndexing.No)
+        switch (_index.CoraxComplexFieldIndexingBehavior)
         {
-            // We need to disable the complex object handling after we check and then throw. 
-            DisableIndexingForComplexObject(field);
-            ThrowIndexingComplexObjectNotSupported(field, _index.Type);
+            case CoraxComplexFieldIndexingBehavior.Throw:
+                ThrowIndexingComplexObjectNotSupportedInStaticIndex(field, _index.Definition.Version);
+                break;
+            case CoraxComplexFieldIndexingBehavior.Skip:
+                // static Corax indexes don't support indexing of complex objects, so we're not going to index it anyway
+                break;
+            default:
+                throw new NotSupportedException(
+                    $"Unknown {nameof(CoraxComplexFieldIndexingBehavior)} option: {_index.CoraxComplexFieldIndexingBehavior}");
         }
+    }
 
-        DisableIndexingForComplexObject(field);
+    private void ComplexObjectInStaticIndexLegacyHandling(IndexField field)
+    {
+        // Backward compatibility for older indexes
+        // Previously, we silently changed the definition not to throw when encountering a complex field without any particular configuration.
 
+        if (IgnoreComplexObjectsDuringIndex) 
+            return;
+
+        Debug.Assert(_index.Type.IsStatic(), "Legacy complex field handling is supposed to be called for static indexes");
+        Debug.Assert(field.Indexing != FieldIndexing.No, "field.Indexing != FieldIndexing.No");
+        Debug.Assert(_index.Definition.Version < IndexDefinitionBaseServerSide.IndexVersion.CoraxComplexFieldIndexingBehavior, "Legacy complex field handling is supposed to be called only for old indexes");
+
+        // Skip indexing of this complex field for current and next entries by overwriting field.Indexing option by FieldIndexing.No
+
+        DisableIndexingForComplexObjectLegacyHandling(field);
+
+        if (_index.GetIndexDefinition().Fields.TryGetValue(field.Name, out var fieldFromDefinition) &&
```

Comment:
This is called only once effectively because we call `DisableIndexingForComplexObjectLegacyHandling()` which sets `field.Indexing = FieldIndexing.No;` so we won't enter here again because we will not attempt to index it again.